src/auth0.ts

```diff
@@ -1,19 +1,31 @@
 import Auth from './auth';
 import CredentialsManager from './credentials-manager';
 import Users from './management/users';
+import {Telemetry} from './networking/telemetry';
 import WebAuth from './webauth';
 
 /**
  * Auth0 for React Native client
  */
 class Auth0 {
+  public auth;
+  public webAuth;
+  public credentialsManager;
+  private options;
+
   /**
    * Creates an instance of Auth0.
    * @param {Object} options your Auth0 application information
    * @param {String} options.domain your Auth0 domain
    * @param {String} options.clientId your Auth0 application client identifier=
    */
-  constructor(options = {}) {
+  constructor(options: {
+    domain: string;
+    clientId: string;
+    telemetry?: Telemetry;
+    token?: string;
+    timeout?: number;
+  }) {
```

Comment:
Is there a reason we do not have an explicit type for the Auth0 client itself? Feels like the first thing users interact with.